backend/src/http/routes/api/questionnaire/questionnaire-routes.js

```diff
@@ -169,7 +169,9 @@ module.exports = ({ db, logger, configuration, regions, communes }) => {
         let stagiaire = req.trainee;
 
         if (stagiaire.avisCreated) {
-            throw new AlreadySentError();
+            res.status(200).send({ stagiaire, infosRegion: await getInfosRegion(stagiaire), alreadySent: true });
```

Comment:
Ca m'embete un peu de retourner une 200 alors qu'il y a une erreur. 
Du coup on ne pourra plus le monitorer (sans alerting).
Est ce que qu'on ne peut pas laisser la 423 et gérer la cas côté front ?

Reply:
ça me pose problème le traitement générique fait côté React dans _post, l'erreur http va générer une exception sur laquelle on n'a pas la main ou alors on utilise fetch en direct pour ce cas. sur le principe j'aurais préféré gardé la 423 également

Reply:
Le `_post` propage l'exception, du coup tu peux avoir le main sur l'erreur dans le Questionnaire.
C'est ce qu'on fait actuellement pour afficher l'erreur lors de la 423

```
try {
    let data = await submitAvis(this.props.stagiaire.token, {
       ...
    });
    this.props.onSubmit(data);
    ...
} catch (ex) {
   // Actuellement on affiche un message d'erreur
    console.error('An error occured', ex);
    this.setState({ showErrorMessage: true });
}
```

Reply:
On pourrait changer la signature de la fonction onSubmit pour la mettre à la mode callback node
`onSubmit(data)` --> `onSubmit(err, data)`

Si err===null alors ok sinon on gère l'erreur.
Du coup ca donnerait un truc genre 

try {
    let data = await submitAvis(this.props.stagiaire.token, {
       ...
    });
    this.props.onSubmit(data);
    ...
} catch (ex) {
   this.props.onSubmit(ex, null);
}

Reply:
oui ok